### PR TITLE
Ensure control boxes covers elements

### PR DESCRIPTION
**Pull request by EivinS**

### CODE REVIEW

File: scripts/drag-n-bar.js [hunk 1]
```diff
@@ -1117,18 +1118,17 @@ H5P.DragNBar.prototype.addControlBoxOnElement = function (element) {
...
 
-    // Moving the control-box to overlap the element.
-    const rectElementBCR = element.$element[0].getBoundingClientRect();
-    const theControlBoxElement = document.getElementsByClassName(uniqueControlBox)[0];
-    const dndToolbarHeight = 38;
-    theControlBoxElement.style.transform = `translate3d(${rectElementBCR.left + window.scrollX}px, ${rectElementBCR.top + window.scrollY + dndToolbarHeight}px, 0px)`;
+    // Since the context-menu-box will be added after the control-box, it will push the element down, but not the control-box-element.
+    // Therefore, we are adjusting the control-box's position on the element 'after' the context-menu has pushed the element down by using requestAnimationFrame(),
+    // so that the control-box is exactly overlapping the element.
+    requestAnimationFrame(() => self.adjustControlBoxPositionOnElement(element, uniqueControlBoxId));
```
Comment (boyum) [inline scripts/drag-n-bar.js:1131]: There's no reason for aliasing `this` here.

```suggestion
    requestAnimationFrame(() => this.adjustControlBoxPositionOnElement(element, uniqueControlBoxId));
```

File: scripts/drag-n-bar.js [hunk 2]
```diff
@@ -1166,15 +1166,27 @@ H5P.DragNBar.prototype.removeControlBoxesNotInUse = function () {
...
-    for (var i = x.length - 1; i >= 0; i--) {
-      x[i].style.visibility = 'hidden';
-    }
+/**
+ *  Hiding moveable-control-boxes.
+ */
+H5P.DragNBar.prototype.hideControlBoxes = function () {
+  var x = document.getElementsByClassName('moveable-control-box');
+  for (var i = x.length - 1; i >= 0; i--) {
+    x[i].style.visibility = 'hidden';
```
Comment (boyum) [inline scripts/drag-n-bar.js:1175]:
```suggestion
H5P.DragNBar.prototype.hideControlBoxes = function () {
  const controlBoxes = document.getElementsByClassName('moveable-control-box');
  for (const controlBox of controlBoxes) {
    controlBox.style.visibility = 'hidden';
```

Comment (EivinS) [reply scripts/drag-n-bar.js:1175]: Sweet, ty!

File: scripts/drag-n-bar.js [hunk 3]
```diff
@@ -1438,6 +1450,13 @@ H5P.DragNBar.prototype.findNewPoint = function (originX, originY, angle, distanc
     }
     frame.rotate = angle;
 
+    // set start transform
+    const transformCSSTranslateXYArray = $element[0].style.transform.split("px");
+    const transformCSSTranslateX = parseInt(transformCSSTranslateXYArray[0].match(/-?\d+/g));
+    const transformCSSTranslateY = parseInt(transformCSSTranslateXYArray[1].match(/-?\d+/g));
```
Comment (boyum) [inline scripts/drag-n-bar.js:1455]: This works, however passing an array into `parseInt` might give unexpected results:

```js
parseInt(["123"]) === 123;
parseInt(["123", "321"]) === 123;
```

I'd prefer to use the first element of the match array:

```suggestion
    const transformCSSTranslateX = parseInt(transformCSSTranslateXYArray[0].match(/-?\d+/g)[0]);
    const transformCSSTranslateY = parseInt(transformCSSTranslateXYArray[1].match(/-?\d+/g)[0]);
```

Comment (EivinS) [reply scripts/drag-n-bar.js:1455]: Aha, TIL

**Review by boyum — APPROVED** [review]

**Pull request merged**